src/components/PersonLink.tsx

```diff
@@ -0,0 +1,56 @@
+import { Link, useLocation, useParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+
+type Props = {
+  person: Person;
+  findParentPerson: (parentName: string) => Person | undefined;
+};
+
+const getLinkClass = (person: Person) => {
+  return classNames({ 'has-text-danger': person.sex === 'f' });
+};
+
+export const PersonLink: React.FC<Props> = ({ person, findParentPerson }) => {
+  const { slug } = useParams();
+  const location = useLocation();
+
+  const renderParentLink = (parentName: string | null) => {
+    const parent = parentName ? findParentPerson(parentName) : undefined;
+
+    return parent ? (
+      <Link
+        className={getLinkClass(parent)}
+        to={`/people/${parent.slug}${location.search}`}
+      >
+        {parent.name}
+      </Link>
+    ) : (
+      parentName || '-'
+    );
+  };
+
+  return (
```

Comment:
It's a good practice to wrap your functional components with React.memo for optimization. It prevents unnecessary renders when the props do not change.